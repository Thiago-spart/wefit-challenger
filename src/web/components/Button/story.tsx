/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import { Button } from "./";

const meta: Meta<typeof Button> = {
	title: "Components/Button",
	component: Button,
};

const Template: ComponentStory<typeof Button> = props => <Button {...props} />;

export const Primary = Template.bind({});
Primary.args = {
	type: "button",
	children: "Click me",
	isFullWidth: false,
	schema: "primary",
	fontSize: "XS",
};

export const Secondary = Template.bind({});
Secondary.args = {
	type: "button",
	children: "Click me",
	isFullWidth: false,
	schema: "secondary",
	fontSize: "XS",
};

export default meta;
