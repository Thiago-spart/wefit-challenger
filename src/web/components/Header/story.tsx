/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import { Header } from "./";

const meta: Meta<typeof Header> = {
	title: "Layout/Header",
	component: Header,
};

const Template: ComponentStory<typeof Header> = props => <Header {...props} />;

export const Default = Template.bind({});
Default.args = {
	cartAmount: 1,
};

export default meta;
