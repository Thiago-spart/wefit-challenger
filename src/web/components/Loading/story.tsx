/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import { Loading } from "./";

const meta: Meta<typeof Loading> = {
	title: "Components/Loading",
	component: Loading,
};

const Template: ComponentStory<typeof Loading> = () => <Loading />;

export const Primary = Template.bind({});
Primary.args = {};

export default meta;
