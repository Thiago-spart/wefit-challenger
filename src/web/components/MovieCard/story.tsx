/* eslint-disable import/no-default-export */
/* eslint-disable import/exports-last */
import type { ComponentStory, Meta } from "@storybook/react";

import { MovieCard } from "./";

const meta: Meta<typeof MovieCard> = {
	title: "Components/MovieCard",
	component: MovieCard,
};

const Template: ComponentStory<typeof MovieCard> = props => (
	<MovieCard {...props} />
);

export const Default = Template.bind({});
Default.args = {
	handleAddProduct: () => {},
	imgAlt: "Some alt description",
	title: "Pele Forever",
	price: 9000,
	productAmount: 2,
	imgLink: "https://www.imagemhost.com.br/images/2022/07/10/the-batman.png",
};

export default meta;
