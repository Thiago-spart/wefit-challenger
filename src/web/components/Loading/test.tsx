import { composeStories } from "@storybook/testing-react";
import { render } from "@testing-library/react";

import * as stories from "./story";

const { Primary } = composeStories(stories);

describe("Loading component", () => {
	it("should match snapshot", () => {
		const { container } = render(<Primary />);

		expect(container).toMatchSnapshot();
	});
});
