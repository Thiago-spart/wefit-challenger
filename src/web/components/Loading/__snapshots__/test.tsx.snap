// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Button component should match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe hCRGab"
  >
    <span
      aria-label="Loading..."
      class="sc-gswNZR jaUCbZ"
    />
  </div>
</div>
`;

exports[`Loading component should match snapshot 1`] = `
<div>
  <div
    class="sc-bcXHqe hCRGab"
  >
    <span
      aria-label="Loading..."
      class="sc-gswNZR jaUCbZ"
    />
  </div>
</div>
`;
